### ISSUE TITLE
binary_cross_entropy crashes when data on mps

### ISSUE THREAD

**Issue by Podidiving** [pytorch/pytorch#81584]
### 🐛 Describe the bug

Hello everyone,

Trying to run the following code

```python
from torch import nn
import torch
from torch.nn import functional as F


device = torch.device("mps")
model = nn.Sequential(
    nn.Linear(10, 1),
    nn.Sigmoid()
)
model.to(device)
input = torch.randn(5, 10).to(device)
target = torch.randint(0, 2, (5, 1)).float().to(device)

F.binary_cross_entropy(
    model(input),
    target
)
```

Expected behaviour is calculated loss. Instead, my interpreter crashes with the following error

```bash
loc("reductionMeanTensor"("(mpsFileLoc): /AppleInternal/Library/BuildRoots/b6051351-c030-11ec-96e9-3e7866fcf3a1/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphUtilities.mm":281:0)): error: invalid axes: 1
```

### Versions

PyTorch version: 1.12.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.13 (main, May 24 2022, 21:13:51)  [Clang 13.1.6 (clang-1316.0.21.2)] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.1
[pip3] torch==1.12.0
[pip3] torchvision==0.13.0
[conda] Could not collect

cc @kulinseth @albanD

**Comment by qqaatw** [comment]
I cannot reproduce this. Can you try the latest nightly build?

**Comment by Podidiving** [comment]
I've tried this with the latest nightly version (`1.13.0.dev20220717`)

Now, the above code throws a warning:

```
/Users/podidiving/env/lib/python3.9/site-packages/torch/_tensor_str.py:103: UserWarning: The operator 'aten::bitwise_and.Tensor_out' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/aten/src/ATen/mps/MPSFallback.mm:11.)
  nonzero_finite_vals = torch.masked_select(tensor_view, torch.isfinite(tensor_view) & tensor_view.ne(0))
```

**Comment by qqaatw** [comment]
The warning is normal. It indicates that some operators haven’t been
supported by MPS and will fallback to CPU implicitly.

**Comment by kulinseth** [comment]
Thanks @qqaatw for trying and providing feedback. I tried locally with latest PyTorch nightly and was not able to reproduce the issue.
